File: mtproto/src/main/java/com/walhalla/mtprotolist/ProtoAdapterCallback.java
package com.walhalla.mtprotolist;


public interface ProtoAdapterCallback<T> {
    void categorySelected(int position, T title);

    void shareProxy(int adapterPosition, T data);

    void qrProxyCode(int adapterPosition, T data);

    void handleProxyIntent(int adapterPosition, T data);

    void viewProxy(int adapterPosition, T data);

    void copyClipboard(String host);

    void delete(T data);

    void info(T data);

    void copyPort(T data);
}

File: mtproto/src/main/java/com/walhalla/mtprotolist/ProtoAdapterCallback.kt
package com.walhalla.mtprotolist


interface ProtoAdapterCallback<T> {
    fun categorySelected(position: Int, title: T)

    fun shareProxy(adapterPosition: Int, data: T)

    fun qrProxyCode(adapterPosition: Int, data: T)

    fun handleProxyIntent(adapterPosition: Int, data: T)

    fun viewProxy(adapterPosition: Int, data: T)

    fun copyClipboard(host: String)

    fun delete(data: T)

    fun info(data: T)

    fun copyPort(data: T)
}